lib/dangermattic/plugins/pr_size_checker.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+module Danger
+  # Plugin to check the size of a Pull Request content and text body.
+  class PRSizeChecker < Plugin
+    DEFAULT_MAX_DIFF_SIZE = 500
+    DEFAULT_DIFF_SIZE_MESSAGE = "This PR is larger than #{DEFAULT_MAX_DIFF_SIZE} lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.".freeze
```

Comment:
Nitpick: the default diff message uses the default max diff size. That is appropriate but might result in inconsistent messages when the `max_size` is set but `message` isn't.

I don't consider this a blocker, but I think we could improve it by using a `Proc` instead:

```ruby
DEFAULT_DIFF_SIZE_MESSAGE = Proc.new { |size| "This PR is larger than #{size} lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews." }
```

_I haven't tested the code_